code/headers/gas_sensors_c.hpp

```diff
@@ -48,16 +48,14 @@ namespace r2d2::gas_detection {
                         container.set(gas.gas_id, gas.value);
                     }
                 }
-
                 if (!frame.request && !has_send) {
                     continue;
                 }
-
                 for (gas_sensor_interface_c<AmountOfGasses> *sensor : sensors) {
                     std::array<r2d2::gas_detection::gas_s, AmountOfGasses>
                         sensor_data = sensor->get();
                     for (r2d2::gas_detection::gas_s gas : sensor_data) {
-                        frame_gas.gas_id = gas.gas_id;
+                        frame_gas.gas_id = uint8_t(gas.gas_id);
```

Comment:
Here you cast the gas type to an uint8_t and in the operator<< for gas you cast it to an int. I think this is a bit inconsistent.

Reply:
This is a c style case. You should use c++ static_cast